modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -516,6 +598,29 @@ cv::GArg IECallContext::packArg(const cv::GArg &arg) {
     }
 }
 
+cv::MediaFrame* IECallContext::prepareKeepAliveFrameSlot(req_key_t key) {
+    std::unique_lock<std::mutex> lock(keep_alive_frames_mutex);
```

Comment:
IMHO:
Probably, 'lock_guard' is a bit more effective, since you don't need to use `lock`/`unlock` directly.
But, again, it's a little thing